File: Java/LeetCode/src/main/java/com/example/leetcode/binarySearch.java
package com.example.leetcode;

/**
 * @author huangchunchen
 * @date 2021/1/6 9:34
 * @description
 */
public class binarySearch {
    public static void main(String[] args) {
        int[] nums = new int[]{1,3,4,5,7,9};
        System.out.println(binarySearch(nums, 11));
    }

    public static int binarySearch(int[] nums, int target){
        int start =0;
        int end = nums.length-1;
        while (start <= end){
            int mid = (start+end)/2;
            if (nums[mid] < target){
                start = mid +1;
            }else if (nums[mid] > target){
                end = mid -1;
            }else {
                return mid;
            }
        }
        return -1;
    }
}

File: go/LeetCode/BinarySearch.go
package main

func main() {
	nums := []int{1,3,4,6,7,9}
	print(binarySearch(nums,4))
}

func binarySearch(nums []int, target int) int {
	left:=0
	right:=len(nums)-1
	for left<=right {
		mid :=left+(right-left)/2
		if nums[mid]==target {
			return mid
		}else if nums[mid]>target {
			right = mid-1
		}else {
			left = mid+1
		}
	}
	return -1
}
